text/0000-async-stream.md

```diff
@@ -91,6 +103,24 @@ The arguments to `poll_next` match that of the [`Future::poll`] method:
 [context]: https://doc.rust-lang.org/std/task/struct.Context.html
 [`Waker`]: https://doc.rust-lang.org/std/task/struct.Waker.html
 
+## next method
+
+As @yoshuawuyts states in their [pull request which adds `core::stream::Stream` to the standard library]:
+
+Unlike `Iterator`, `Stream` makes a distinction between the [`poll_next`]
+method which is used when implementing a `Stream`, and the [`next`] method
+which is used when consuming a stream. Consumers of `Stream` only need to
+consider [`next`], which when called, returns a future which yields
+[`Option`]`<Item>`.
+
+The future returned by [`next`] will yield `Some(Item)` as long as there are
+elements, and once they've all been exhausted, will yield `None` to indicate
+that iteration is finished. If we're waiting on something asynchronous to
+resolve, the future will wait until the stream is ready to yield again.
+
+Individual streams may choose to resume iteration, and so calling [`next`]
+again may or may not eventually yield `Some(Item)` again at some point.
```

Comment:
We could possibly highlight a similarity with `Future` here too: the `Future::poll_next` trait by itself is rarely called directly — it's only ever really used to implement other futures. Instead interacting with futures needs to either be done `async/.await`.

Similarly `Stream` by itself is not useful — we need some way to interact with it. And the foundational interaction is exactly what `next` provides: all other interactions with stream can be expressed through it. But without it streams can't be consumed.

Reply:
Will add that, thank you!